src/pathfinder.rs

```diff
@@ -423,11 +423,33 @@ where
                 heuristicWeight: @{heuristic_weight}
             })
         };
+        //In some cases, like if we are searching from a position with no walkable neighbors,
+        //js comes back with `null` here.
+        //For reasons opaque to me, this is bridged into f64::POSITIVE_INFINITY rather than stdweb::Value::Null
+        let cost: u32 = match js! {return @{&res}.cost;} {
+            stdweb::Value::Number(n) => {
+                //see if the number fits into u32
+                if let Some(u) = n.try_into().ok() {
+                    u
+                } else if let Some(f) = TryInto::<f64>::try_into(n).ok() {
+                    if f.is_infinite() {
+                        std::u32::MAX
```

Comment:
Any reasons not to switch to an `Option<u32>`? 

In any case, if we go this route we'd need to document this somewhere as someone might still try to follow that path if the cost is a valid one.